src/main/resources/assets/js/components/Marathon.jsx

```diff
@@ -78,26 +135,31 @@ var Marathon = React.createClass({
     }.bind(this));
 
     Mousetrap.bind("shift+,", function () {
-      this.showAboutModal();
+      router.navigate("#about", {trigger: true});
     }.bind(this));
 
-    this.setPollResource(this.fetchApps);
+    this.updatePolling();
   },
 
   componentDidUpdate: function (prevProps, prevState) {
-    if (prevState.modalClass !== this.state.modalClass) {
-      // No `modalClass` means the modal went from open to closed. Start
-      // polling for apps in that case.
-      // If `modalClass` is AppModalComponent start polling for tasks for that
-      // app.
-      // Otherwise stop polling since the modal went from closed to open.
-      if (this.state.modalClass === null) {
-        this.setPollResource(this.fetchApps);
-      } else if (this.state.modalClass === AppModalComponent) {
-        this.setPollResource(this.fetchTasks);
-      } else {
-        this.stopPolling();
-      }
+    /* jshint eqeqeq: false */
+    if (prevState.activeApp != this.state.activeApp) {
+      this.updatePolling();
+    }
+
+    if (this.state.activeAppId) {
+      this.state.setAppView(this.state.activeAppId, this.state.activeAppView);
+    }
+
+    var route = this.state.route;
+    var router = this.props.router;
+
+    if (route) {
```

Comment:
This is a little bit tricky. We need this to populate the tasks for the AppModal correctly. Because the AppModal can be opened on various states.